.github/workflows/kiali-release.yml

```diff
@@ -0,0 +1,117 @@
+        type: string
+        default: quay.io/leandroberetta/kiali
+        required: true 
+
+jobs:
+  # The initialize job gathers variables for later use in jobs.
+  # We are using this technique rather environment variables because at the moment, they won't work with reusable jobs.  
+  initialize:
+    name: Initialize
+    runs-on: ubuntu-latest
+    outputs:
+      target_branch: ${{ steps.target_branch.outputs.target_branch }}
+      release_type: ${{ steps.release_type.outputs.release_type || 'auto' }}      
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - id: target_branch
+        run: echo "::set-output name=target_branch::${{ github.base_ref || github.ref_name }}"        
+      - name: Determine release type
+        id: release_type
+        if: ${{ github.event.inputs.release_type == 'auto'}}
+        run: echo "::set-output name=release_type::$(./.github/workflows/util/release_type.sh)"
+  build_backend:
+    name: Build backend
+    uses: ./.github/workflows/build-backend.yml
+    needs: [initialize]
+  build_frontend:
+    name: Build frontend
+    uses: ./.github/workflows/build-frontend.yml
+    needs: [initialize]
+    with:
+      target_branch: ${{needs.initialize.outputs.target_branch}}
+  integration_tests_backend:
+    name: Run backend integration tests
+    uses: ./.github/workflows/integration-tests-backend.yml
+    needs: [initialize, build_backend, build_frontend]
+    with:
+      target_branch: ${{needs.initialize.outputs.target_branch}}
+  integration_tests_frontend:
+    name: Run frontend integration tests
+    uses: ./.github/workflows/integration-tests-frontend.yml
+    needs: [initialize, build_backend, build_frontend]
+    with:
+      target_branch: ${{needs.initialize.outputs.target_branch}}
+  release_kiali:
+    name: Release Kiali
+    runs-on: ubuntu-20.04
+    needs: [initialize, build_backend, build_frontend, integration_tests_backend, integration_tests_frontend]
+    env:
+      RELEASE_TYPE: ${{ needs.initialize.outputs.release_type}}        
+      BACKEND_REPO: ${{ github.event.inputs.backend_repo }}
+      QUAY_NAME: ${{ github.event.inputs.quay_name }}
+    steps:    
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.7
+
+      - name: Cache Go deps
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Download frontend build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+          path: frontend/build       
+
+      - name: Build backend for release (Multi-arch)      
+        run: make -f ./.github/workflows/util/Makefile backend-build-release
```

Comment:
I think you no longer need this step, since you have previously ran the `build-backend.yml` workflow. It is redundant. Perhaps, simply unify the `sed` commands in this make-target into the `backend-push-docker`.

The important point is that you should not build the back-end yet again (just for saving some time). The next step (Push Kiali to Quay) is the one that is doing the multi-arch builds.